### PR TITLE
[License Checker] Add license checker script && apply it on postinstall

### PR DESCRIPTION
**Pull request by AtticusFetch**
## PR Type

PR introduces the following changes:

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe:
```

Add license checker script to verify added direct dependencies against allowed open source licenses.

## Actual/New behavior

On postinstall stage, a script will be executed that will verify that all the dependencies licenses are in the allowed list of open-source licenses. If not - script will throw an error an the process will fail. In addition it will create a JSON file containing dependencies license info that we can display in our app to meet the open source licenses requirements

### CODE REVIEW

File: scripts/acknowledgements.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      }
+      delete libData.publisher;
+      delete libData.email;
+      delete libData.url;
+      delete libData.path;
+      const libInfo = {
+        ...libData,
+        license: fs.readFileSync(strippedVersions[lib].licenseFile, { encoding: 'UTF-8' }),
+        name: lib,
+      };
```
Comment (kovnick) [inline scripts/acknowledgements.js]: Do we need full license text? Probably we could store only licenses/repository/name fields.

File: scripts/acknowledgements.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      if (!libInfo) {
+        console.error('not found');
+      } else {
+        acknowledgements.push(libInfo);
+      }
+    });
+
+    if (acknowledgements.length) {
+      fs.writeFileSync('./acknowledgements.json', JSON.stringify(acknowledgements, null, 2));
```
Comment (kovnick) [inline scripts/acknowledgements.js]: Should we store these acknowledgements.json files? Throwing error not enough?

File: client/acknowledgements.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[
+  {
+    "licenses": "MIT",
+    "repository": "https://github.com/axios/axios",
+    "dependencyPath": "/Users/ivaniankovskyi/Projects/questionator/client/node_modules/axios",
```
[the remark this answers is not in the document]
Comment (AtticusFetch) [reply client/acknowledgements.json]: Ooops

File: scripts/acknowledgements.js (hunk 1 above)
Comment (AtticusFetch) [reply scripts/acknowledgements.js]: The thing is some of the licenses even though are open-source require users to distribute license text as well. So, we should have some sort of acknowledgements sections with licenses text of our dependencies in there.

This is the only reason for this.

File: scripts/acknowledgements.js (hunk 2 above)
Comment (AtticusFetch) [reply scripts/acknowledgements.js]: same as above. File could be used for the acknowledgements page

File: scripts/acknowledgements.js (hunk 1 above)
Comment (AtticusFetch) [reply scripts/acknowledgements.js]: e.g.:

> The above copyright notice and this permission notice shall be\nincluded in all copies or substantial portions of the Software.

**Pull request merged**